test/functional/mining_basic.py

```diff
@@ -336,7 +337,7 @@ def test_block_max_weight(self):
 
         self.log.info("Test -blockreservedweight startup option.")
         # Lowering the -blockreservedweight by 4000 will allow for two more transactions.
-        self.restart_node(0, extra_args=[f"-datacarriersize={LARGE_VSIZE}", "-blockreservedweight=4000"])
+        self.restart_node(0, extra_args=[f"-datacarriersize={LARGE_VSIZE}", "-blockreservedweight=4000"], expected_stderr=deprecation_warning)
```

Comment:
This looked kind of confusing to me: The node is restarted without `blockmaxweight` but the deprecation warning from init is still expected to show here. Looking at how this works in `restart_node` explains why: The warning is checked during stopping of the node and not during starting of node. I think it would be good to add at least some comment here how this works. Maybe there is also a more expressive way to handle it but I think a comment is good enough.